src/gui/ScrollHelpers.cpp

```diff
@@ -0,0 +1,90 @@
+/*
+ * ScrollHelpers.cpp - helper functions for wheel events
+ *
+ * Copyright (c) 2023 Alex <allejok96/gmail>
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#include "ScrollHelpers.h"
+
+#include <QWheelEvent>
+
+
+namespace lmms {
+
+
+bool ignoreScroll(const Qt::Orientation orientation, QWheelEvent* event)
+{
+	bool hasOtherOrientation = orientation == Qt::Horizontal ? event->angleDelta().y() : event->angleDelta().x();
+	event->setAccepted(hasOtherOrientation);
+	return !hasOtherOrientation;
+}
+
+
+
+
+// TODO: is there a good way to prevent calling this method multiple times with the same event?
+int getScroll(const QWheelEvent* event, const Qt::Orientation orientation, const float factor, const bool allowNatural)
+{
+	static int xRemainder;
+	static int yRemainder;
```

Comment:
I'm not a huge fan of these being static. The value will carry over to another event and might cause unintended scroll values.

Reply:
I'm not sure, but per-widget counters might be useful for fractional parts.

Reply:
I've considered this a lot... My conclusion is: does it really matter?

Per-widget counter - the logical solution:
1. Scroll knob A 90% of a step - nothing happens.
2. Scroll knob B.
3. Scroll knob A slightly - it turns over.
4. User is surprised, because it's impossible to keep track of the scroll remainder of every knob.

Global counter - the simple solution:
1. Scroll knob A 90% of a step - nothing happens.
3. Scroll knob B slightly - it turns over.
5. User is surprised, because it's impossible to keep track of the scroll remainder of every knob.

Another thing: the remainder is reset every time the wheel changes direction. This is how Qt handles it and it's a clever way to prevent offsets building up over time and also pretty logical when you think about it. But it comes with this side effect:

1. Scroll up 90% of a step - nothing happens.
2. Scroll down 10% - nothing happens.
3. Scroll up 90% - nothing happens.